spec/system/banners/banners_spec.rb

```diff
@@ -0,0 +1,173 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+include UsersHelper
+
+RSpec.describe 'Banners', type: :system do
+  describe 'Index', js: true do
+    before(:each) do
+      @banner             = create(:banner)
+      visit root_path
+    end
+
+
+
+    describe 'user should not have access' do
+      before(:each) do
+        @user  = create(:user)
+        login_as @user
+      end
+
+      context '#index' do
+        scenario 'displays proper error message for non-admin user' do
+          visit banners_path
+          expect(current_path).to eq("/")
+          expect(page).to have_content('You are not authorized to perform this action.')
+        end
+      end
+    end
+
+    describe 'admin user' do
+      before(:each) do
+        @system_admin_user  = create(:system_admin_user)
+        login_as @system_admin_user
+
+        visit root_path
+      end
+
+      context 'home page' do
+        context 'header links' do
+          scenario 'displays login link' do
+            expect(page).to have_link 'Admin'
+          end
+
+          scenario 'displays the Banners link for admins' do
+            click_link 'Admin'
+            click_link 'Banners'
+            expect(current_path).to eq("/banners")
```

Comment:
This and other admin paths (All Users, All Grants) could be moved into an admin namespace. (admin/banners, admin/users). Will discuss and open a ticket about it.

Reply:
I like this idea, but it should be a new ticket and planned into a future sprint.